src/routes/transactions/transactions-view.service.ts

```diff
@@ -20,9 +32,78 @@ export class TransactionsViewService {
       to: args.transactionDataDto.to,
     });
 
-    return new ConfirmationView({
-      method: dataDecoded.method,
-      parameters: dataDecoded.parameters,
+    const swapOrderData = this.swapOrderHelper.findSwapOrder(
+      args.transactionDataDto.data,
+    );
+
+    if (!swapOrderData) {
+      return new BaselineConfirmationView({
+        method: dataDecoded.method,
+        parameters: dataDecoded.parameters,
+      });
+    }
+
+    try {
+      return await this.handleSwapOrder({
```

Comment:
Nit: can we rename this to `getSwapOrderConfirmationView` or something similar? I was a bit confused on face value that it was "handlin" but not returning something.

Reply:
Done in [59ff543](https://github.com/safe-global/safe-client-gateway/pull/1435/commits/59ff543211fc6f4cdabc35c163e23fb33c9c7aa1)